support/metrics/metricsTracker.go

```diff
@@ -190,34 +188,65 @@ func MakeMetricsTracker(
 	}, nil
 }
 
+// MakeMetricsTrackerGui is a factory method to create a `metrics.Tracker` from the CLI.
+func MakeMetricsTrackerGui(
+	userID string,
+	deviceID string,
+	apiKey string,
+	client *http.Client,
+	botStartTime time.Time,
+	version string,
+	goos string,
+	goarch string,
+	goarm string,
+	guiVersion string,
+) (*MetricsTracker, error) {
+	props := commonProps{
+		CliVersion: version,
+		Goos:       goos,
+		Goarch:     goarch,
+		Goarm:      goarm,
+		GuiVersion: guiVersion,
+	}
+
+	return &MetricsTracker{
+		client:       client,
+		apiKey:       apiKey,
+		userID:       userID,
+		deviceID:     deviceID,
+		props:        props,
+		botStartTime: botStartTime,
+	}, nil
+}
+
 // GetUpdateEventSentTime gets the last sent time of the update event.
 func (mt *MetricsTracker) GetUpdateEventSentTime() *time.Time {
 	return mt.updateEventSentTime
 }
 
 // SendStartupEvent sends the startup Amplitude event.
 func (mt *MetricsTracker) SendStartupEvent() error {
-	return mt.sendEvent(startupEventName, mt.props)
+	return mt.SendEvent(startupEventName, mt.props)
```

Comment:
@debnil how we can convert SendEvent's second param to map[string]interface{}
```
v, e := toMapStringInterface(mt.props)
// check e
return mt.SendEvent(startupEventName, v)

func toMapStringInterface(v interface{}) (map[string]interface{}, error) {
        var m map[string]interface{}
	eventJSON, e := json.Marshal(v)
	if e != nil {
		return nil, fmt.Errorf("could not marshal to map[string]interface{}: %s", e)
	}
}
```

Reply:
I don't think this method quite works, since `json.Marshal` creates a `[]byte`. To create a `map[string]interface{}`, I would then have to call `json.Unmarshal` on the resulting byte-array - which is the exact approach we tried to avoid.

I think all we need to do is some syntactic sugar around a typecast to `map[string]interface{}`, instead, which I've added in `toMapStringInterface`. This checks the `ok` boolean and returns an error if false.